qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/MSDFunctionModel.cpp

```diff
@@ -0,0 +1,520 @@
+  out[static_cast<int>(ParamID::GAUSSIAN_HEIGHT)] =
+      gaussian->parameterDescription(0);
+  out[static_cast<int>(ParamID::GAUSSIAN_MSD)] =
+      gaussian->parameterDescription(1);
+  auto peters = FunctionFactory::Instance().createInitialized(
+      buildPetersFunctionString());
+  out[static_cast<int>(ParamID::PETERS_HEIGHT)] =
+      peters->parameterDescription(0);
+  out[static_cast<int>(ParamID::PETERS_MSD)] = peters->parameterDescription(1);
+  out[static_cast<int>(ParamID::PETERS_BETA)] = peters->parameterDescription(2);
+  auto yi =
+      FunctionFactory::Instance().createInitialized(buildYiFunctionString());
+  out[static_cast<int>(ParamID::YI_HEIGHT)] = yi->parameterDescription(0);
+  out[static_cast<int>(ParamID::YI_MSD)] = yi->parameterDescription(1);
+  out[static_cast<int>(ParamID::YI_SIGMA)] = yi->parameterDescription(2);
+  return out;
+}
+
+void MSDFunctionModel::setCurrentValues(const QMap<ParamID, double> &values) {
+  for (const auto &name : values.keys()) {
+    setParameter(name, values[name]);
+  }
+}
+
+void MSDFunctionModel::applyParameterFunction(
+    std::function<void(ParamID)> paramFun) const {
+  if (m_fitType == "MsdGauss") {
+    paramFun(ParamID::GAUSSIAN_HEIGHT);
+    paramFun(ParamID::GAUSSIAN_MSD);
+  }
+  if (m_fitType == "MsdPeters") {
+    paramFun(ParamID::PETERS_HEIGHT);
+    paramFun(ParamID::PETERS_MSD);
+    paramFun(ParamID::PETERS_BETA);
+  }
+  if (m_fitType == "MsdYi") {
+    paramFun(ParamID::YI_HEIGHT);
+    paramFun(ParamID::YI_MSD);
+    paramFun(ParamID::YI_SIGMA);
+  }
+}
+
+boost::optional<MSDFunctionModel::ParamID>
+MSDFunctionModel::getParameterId(const QString &parName) {
+  boost::optional<ParamID> result;
+  auto getter = [&result, parName, this](ParamID pid) {
+    if (parName == *getParameterName(pid))
+      result = pid;
+  };
+  applyParameterFunction(getter);
+  return result;
+}
+
+std::string MSDFunctionModel::buildGaussianFunctionString() const {
+  return "name=MsdGauss,Height=1,Msd=0.05,constraints=(Height>0)";
+}
+
+std::string MSDFunctionModel::buildPetersFunctionString() const {
+  return "name=MsdPeters,Height=1,Msd=0.05,Beta=1,constraints=(Height>0)";
+}
+
+std::string MSDFunctionModel::buildYiFunctionString() const {
+  return "name=MsdYi,Height=1,Msd=0.05,Sigma=1,constraints=(Height>0)";
+}
+
+QString MSDFunctionModel::buildFunctionString() const {
+  QStringList functions;
+  if (m_fitType == "MsdGauss") {
+    functions << QString::fromStdString(buildGaussianFunctionString());
+  }
+  if (m_fitType == "MsdPeters") {
+    functions << QString::fromStdString(buildPetersFunctionString());
+  }
+  if (m_fitType == "MsdYi") {
+    functions << QString::fromStdString(buildYiFunctionString());
+  }
+  return functions.join(";");
+}
+
+boost::optional<QString> MSDFunctionModel::getFitTypePrefix() const {
+  if (m_fitType.isEmpty())
+    return boost::optional<QString>();
+  return QString();
+}
+
+QString MSDFunctionModel::setBackgroundA0(double) { return ""; }
```

Comment:
If this does nothing, could we just delete it and replace the call sites with just setting ""

Reply:
I think this exists to fullfil an Abstract interface requirement rather than to actually do anything in this case. Which I know is not great but would be difficult to remove.